File: net/Link.Message.Client/content/complex/linear/FormContent.cs
using System.Collections.Generic;
using Newtonsoft.Json;

namespace Link.Message.Client.content.complex.linear
{
	public class FormContent : IStringOrObjectSerializable
	{
	    public FormFieldConfig Key { get; set; }
	    public FormFieldConfig Value { get; set; }
	    protected internal IList<FormFieldValue> datas = new List<FormFieldValue>();

		public FormContent(FormFieldConfig key, FormFieldConfig value)
		{
			Key = key;
			Value = value;
		}

        [JsonProperty("data")]
		public virtual IList<FormFieldValue> Datas
		{
			get
			{
				return datas;
			}
		}

		public virtual void AddData(FormFieldValue data)
		{
			datas.Add(data);
		}

		public virtual void AddData(string key, string value)
		{
			datas.Add(new FormFieldValue(key, value));
		}
	}

}

File: java/link-message-client/src/main/java/link/message/client/content/complex/linear/FormContent.java
package link.message.client.content.complex.linear;

import java.util.ArrayList;
import java.util.List;

import link.message.client.StringOrObjectSerializable;

import com.alibaba.fastjson.annotation.JSONField;

public class FormContent implements StringOrObjectSerializable {
	protected FormFieldConfig       key;
	protected FormFieldConfig       value;
	@JSONField(name="data")
	protected List<FormFieldValue>  datas = new ArrayList<FormFieldValue>();
	
	public FormContent(FormFieldConfig key, FormFieldConfig value) {
		this.key   = key;
		this.value = value;
	}

	public FormFieldConfig getKey() {
		return key;
	}

	public void setKey(FormFieldConfig key) {
		this.key = key;
	}

	public FormFieldConfig getValue() {
		return value;
	}

	public void setValue(FormFieldConfig value) {
		this.value = value;
	}

	public List<FormFieldValue> getDatas() {
		return datas;
	}
	
	public void addData(FormFieldValue data) {
		this.datas.add(data);
	}
	
	public void addData(String key, String value) {
		this.datas.add(new FormFieldValue(key, value));
	}
}
